api/py/ai/chronon/repo/compile.py

```diff
@@ -321,7 +321,7 @@ def check_deprecation_existence_in_upstream(collection, entity_type, obj_name):
 def _handle_deprecation_warning(
     obj: Union[Join, GroupBy], obj_class: Type[Union[Join, GroupBy]], downstream_group_bys: dict, downstream_joins: dict
 ) -> None:
-    if obj.metaData.deprecationDate:
+    if obj.metaData.deprecationDate and obj_class in [Join, GroupBy]:
```

Comment:
nit:

```
if not obj.metaData.deprecationDate or obj_class not in [Join, GroupBy]:
    return
```

Reply:
removed this line since the downstream_group_bys and downstream_joins are empty for staging queries now so nothing will be checked. And in the future the downstream for staging queries should be checked as well.